internal/os/os_test.go

```diff
@@ -0,0 +1,80 @@
+package os
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_OperatingSystem(t *testing.T) {
+	subCommand := "operating-systems"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	type args struct {
+		name string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "NoArgs",
+			fields: fields{
+				MainCmd:  &cobra.Command{},
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			args: args{name: "testCommand"},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				root.SetArgs([]string{subCommand})
+				if err := root.Execute(); err != nil {
+					t.Logf("%+v", root.Args)
+					t.Error("expected an error")
+				}
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			args: args{name: "testCommand"},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get"})
+				if err := root.Execute(); err != nil {
+					t.Logf("%+v", root.Args)
+					t.Error("expected an error")
+				}
+				if !strings.Contains(out.String(), "Information on available operating systems") {
+					t.Error("expected output should include list all known hardware")
+				}
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rootCmd := rootClient.NewCommand()
+			rootCmd.AddCommand(tt.fields.MainCmd)
```

Comment:
We need to call the `cmdFunc` to run each test

```suggestion
			rootCmd.AddCommand(tt.fields.MainCmd)
			tt.cmdFunc(t, tt.fields.MainCmd)
```